spark/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/CometShuffleExchangeExec.scala

```diff
@@ -482,8 +483,10 @@ class CometShuffleWriteProcessor(
       "elapsed_compute" -> metrics("shuffleReadElapsedCompute"))
     val nativeMetrics = CometMetricNode(nativeSQLMetrics)
 
-    val rawIter = cometRDD.iterator(partition, context)
-    val cometIter = CometExec.getCometIterator(Seq(rawIter), nativePlan, nativeMetrics)
+    val cometIter = CometExec.getCometIterator(
+      Seq(inputs.asInstanceOf[Iterator[ColumnarBatch]]),
```

Comment:
I think `inputs` is not `Iterator[ColumnarBatch]`. Maybe this will cause runtime exception.

Reply:
We have `ShimCometShuffleWriteProcessor.write()` for Spark 3.x

Yes, currently this may not be correct for Spark 4.0. This will be followup with test fixes. I will ask for your help about how to work around the interface change (or change Spark in the upstream).